lms/static/js/ccx/schedule.js

```diff
@@ -64,11 +67,16 @@ var edx = edx || {};
 		  .append(self.schedule_options(chapter.children));
 		self.sequential_select.prop('disabled', false);
 		$('#add-unit-button').prop('disabled', false);
-		self.set_datetime('start', chapter.start);
-		self.set_datetime('due', chapter.due);
+		// When a chapter is selected, start date fields are enabled and due date
+		// fields are disabled because due dates are not applicable on a chapter.
+		self.disabled_fields($('.ccx_due_date_time_fields'), true);
+		self.disabled_fields($('.ccx_start_date_time_fields'), false);
 	      }
 	      else {
 		self.sequential_select.html('').prop('disabled', true);
+		// When no chapter is selected, all date fields are disabled.
```

Comment:
Shouldn't this logic go into the template, rather than rendering the template and then mutating the HTML based on the data?